### PR TITLE
docs: Move almost everything out of the README and into docs/

### PR DESCRIPTION
**Pull request by timmc-edx**
The readme was too large and was focused on setup, which meant that people weren't aware of other important files such as workflow.rst. Breaking out the contents into separate files will help ensure people actually visit the (easier to navigate, better discoverability) Read The Docs site.

Addresses https://github.com/openedx/devstack/issues/1030

----

Author's note: Commits best reviewed individually.

----

I've completed each of the following or determined they are not applicable:

- [x] Made a plan to communicate any major developer interface changes (or N/A)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -38,13 +38,8 @@ You can also browse all the documentation in `Read the Docs`_.
...
 
-**NOTE:** LMS is now using MySql 5.7 by default. You have to run ``make dev.pull.lms`` and ``make dev.provision.lms`` (more details in `Getting Started`_) to fetch latest images and reprovision local copies of databases in order for an existing devstack setup to keep working.
+**NOTE:** LMS is now using MySql 5.7 by default. You have to run ``make dev.pull.lms`` and ``make dev.provision.lms`` (more details in :doc:`getting_started`) to fetch latest images and reprovision local copies of databases in order for an existing devstack setup to keep working.
 
-.. _Docker Compose: https://docs.docker.com/compose/
-.. _Docker for Mac: https://docs.docker.com/docker-for-mac/
-.. _licensing terms: https://www.docker.com/pricing/faq
-.. _Docker for Windows: https://docs.docker.com/docker-for-windows/
-.. _configuring Docker for Mac: https://docs.docker.com/docker-for-mac/#/advanced
 .. _feature added in Docker 17.05: https://github.com/openedx/configuration/pull/3864
```
Comment (rgraber) [inline README.rst:43]: [question] It seems like a bunch of these links are also unused. Wondering if we should remove them while we're here.

**Review by rgraber — COMMENTED** [review]
one question

File: README.rst (hunk 1 above)
Comment (timmc-edx) [reply README.rst:43]: I was hoping to clean those up in a later PR, if that's alright. (I'm actually a little surprised that sphinx or doc8—or whatever the tooling is—doesn't lint for that. Something does detect unused _files_, at least.)

Comment (timmc-edx) [reply README.rst:43]: I'm also not happy with the placement of the chunk of text I stuck at the top of index.rst, but I think that would be better discussed in a separate PR as well, one that moves less text around.

**Review by rgraber — APPROVED** [review]
LGTM

**Comment by timmc-edx** [comment]
I merged without waiting for provisioning tests to pass, since this is a docs-only PR. Actually, we don't have any CI checks on docs! I'll probably put in a small PR to run `make docs` as a CI check.

**Pull request merged**